src/main/java/be/seeseemelk/mockbukkit/block/data/WallMock.java

```diff
@@ -0,0 +1,116 @@
+package be.seeseemelk.mockbukkit.block.data;
+
+import com.google.common.base.Preconditions;
+import org.bukkit.Material;
+import org.bukkit.Tag;
+import org.bukkit.block.BlockFace;
+import org.bukkit.block.data.type.Wall;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Set;
+
+import static be.seeseemelk.mockbukkit.block.data.BlockDataKey.*;
+
+/**
+ * Mock implementation of {@link Wall}.
+ */
+public class WallMock extends BlockDataMock implements Wall
+{
+
+	/**
+	 * Constructs a new {@link WallMock} for the provided {@link Material}.
+	 * Only supports materials in {@link Tag#WALLS}
+	 *
+	 * @param type The material this data is for.
+	 */
+	public WallMock(@NotNull Material type)
+	{
+		super(type);
+		checkType(type, Tag.WALLS);
+		setHeight(BlockFace.EAST, Height.NONE);
+		setHeight(BlockFace.WEST, Height.NONE);
+		setHeight(BlockFace.NORTH, Height.NONE);
+		setHeight(BlockFace.SOUTH, Height.NONE);
+		setUp(false);
+		setWaterlogged(false);
+	}
+
+	/**
+	 * @return the set of possible block faces for {@link Wall}
+	 */
+	public @NotNull Set<BlockFace> getFaces()
+	{
+		return Set.of(BlockFace.EAST, BlockFace.NORTH, BlockFace.WEST, BlockFace.SOUTH);
```

Comment:
I feel like this could/should be a `private static final` field instead.
No need to instantiate this on every method call or even for every instance of this class.